### PR TITLE
Snowflake and Core OAuth

### LINKED ISSUE #2892: Describe connecting to Snowflake via OAuth
### Contributions

- [X] I have read the contribution docs, and understand what's expected of me.

### Link to the page on docs.getdbt.com requiring updates

https://docs.getdbt.com/reference/warehouse-setups/snowflake-setup#authentication-methods

### What part(s) of the page would you like to see updated?

Describe the OAuth method(s) for connecting to Snowflake.

Use [this](https://github.com/dbt-labs/dbt-snowflake/issues/474#issuecomment-1435368808) content for the crucial parameters.

BigQuery has an analogous OAuth option & instructions [here](https://docs.getdbt.com/reference/warehouse-setups/bigquery-setup#oauth-token-based) which can be used for context and inspiration.

### Additional information

Snowflake Admins will use instructions like [this](https://community.snowflake.com/s/article/HOW-TO-OAUTH-TOKEN-GENERATION-USING-SNOWFLAKE-CUSTOM-OAUTH) and [this](https://docs.getdbt.com/docs/collaborate/manage-access/set-up-snowflake-oauth) to setup this connection method.

But we want to cover how end users can configuration their connection in their dbt `profiles.yml` once their admin has set it up.

### PR DESCRIPTION
**Pull request by matthewshaver**
## What are you changing in this pull request and why?

Adding OAuth configs for Core/Snowflake

Closes https://github.com/dbt-labs/docs.getdbt.com/issues/2892

## Checklist
- [ ] I have reviewed the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) so my content adheres to these guidelines.
- [ ] The topic I'm writing about is for specific dbt version(s) and I have versioned it according to the [version a whole page](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) and/or [version a block of content](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-blocks-of-content) guidelines.
- [ ] I have added checklist item(s) to this list for anything anything that needs to happen before this PR is merged, such as "needs technical review" or "change base branch."
- [ ] The content in this PR requires a dbt release note, so I added one to the [release notes page](https://docs.getdbt.com/docs/dbt-versions/dbt-cloud-release-notes).

---
🚀 Deployment available! Here are the direct links to the updated files:

- https://docs-getdbt-com-git-snowflake-oauth-dbt-labs.vercel.app/docs/core/connect-data-platform/snowflake-setup

### CODE REVIEW

File: website/docs/docs/core/connect-data-platform/snowflake-setup.md [hunk 1]
```diff
@@ -421,6 +421,28 @@ my-snowflake-db:
 
 Refer to the [Snowflake docs](https://docs.snowflake.com/en/sql-reference/parameters.html#label-allow-id-token) for info on how to enable this feature in your account.
 
+### OAuth authorization
+
+See the Snowflake [docs](https://docs.snowflake.com/en/user-guide/oauth-snowflake-overview) for information about configuring OAuth. Your Snowflake admin needs to generate an [OAuth token](https://community.snowflake.com/s/article/HOW-TO-OAUTH-TOKEN-GENERATION-USING-SNOWFLAKE-CUSTOM-OAUTH) for your configuration to work.
+
+The OAUTH_REDIRECT_URI you provide in Snowflake is `http://localhost`. Once Snowflake is configured, add the following to your `profiles.yml` file:
```
Comment (matthewshaver) [inline website/docs/docs/core/connect-data-platform/snowflake-setup.md]:
```suggestion
The OAUTH_REDIRECT_URI you provide in Snowflake is `http://localhost:PORT_NUMBER`. For example, `http://localhost:8080`.

Once your Snowflake admin has configured OAuth, add the following to your `profiles.yml` file:
```

File: website/docs/docs/core/connect-data-platform/snowflake-setup.md [hunk 2]
```diff
@@ -421,6 +421,30 @@ my-snowflake-db:
 
 Refer to the [Snowflake docs](https://docs.snowflake.com/en/sql-reference/parameters.html#label-allow-id-token) for info on how to enable this feature in your account.
 
+### OAuth authorization
+
+See the Snowflake [docs](https://docs.snowflake.com/en/user-guide/oauth-snowflake-overview) for information about configuring OAuth. Your Snowflake admin needs to generate an [OAuth token](https://community.snowflake.com/s/article/HOW-TO-OAUTH-TOKEN-GENERATION-USING-SNOWFLAKE-CUSTOM-OAUTH) for your configuration to work.
```
Comment (matthewshaver) [inline website/docs/docs/core/connect-data-platform/snowflake-setup.md]:
```suggestion
See the Snowflake [docs](https://docs.snowflake.com/en/user-guide/oauth-snowflake-overview) for configuring OAuth. Your Snowflake admin needs to generate an [OAuth token](https://community.snowflake.com/s/article/HOW-TO-OAUTH-TOKEN-GENERATION-USING-SNOWFLAKE-CUSTOM-OAUTH) for your configuration to work.
```

File: website/docs/docs/core/connect-data-platform/snowflake-setup.md [hunk 3]
```diff
@@ -421,6 +421,30 @@ my-snowflake-db:
 
 Refer to the [Snowflake docs](https://docs.snowflake.com/en/sql-reference/parameters.html#label-allow-id-token) for info on how to enable this feature in your account.
 
+### OAuth authorization
+
+See the Snowflake [docs](https://docs.snowflake.com/en/user-guide/oauth-snowflake-overview) for configuring OAuth. Your Snowflake admin needs to generate an [OAuth token](https://community.snowflake.com/s/article/HOW-TO-OAUTH-TOKEN-GENERATION-USING-SNOWFLAKE-CUSTOM-OAUTH) for your configuration to work.
```
Comment (runleonarun) [inline website/docs/docs/core/connect-data-platform/snowflake-setup.md]: Maybe make this active and say something like:
```suggestion
To learn how to configure OAuth in Snowflake, refer to their [documentation](https://docs.snowflake.com/en/user-guide/oauth-snowflake-overview). Your Snowflake admin needs to generate an [OAuth token](https://community.snowflake.com/s/article/HOW-TO-OAUTH-TOKEN-GENERATION-USING-SNOWFLAKE-CUSTOM-OAUTH) for your configuration to work.
```

File: website/docs/docs/core/connect-data-platform/snowflake-setup.md [hunk 4]
```diff
@@ -421,6 +421,30 @@ my-snowflake-db:
...
+
+The OAUTH_REDIRECT_URI you provide in Snowflake is `http://localhost:PORT_NUMBER`. For example, `http://localhost:8080`.
```
Comment (runleonarun) [inline website/docs/docs/core/connect-data-platform/snowflake-setup.md]:
```suggestion
Provide the OAUTH_REDIRECT_URI in Snowflake:`http://localhost:PORT_NUMBER`. For example, `http://localhost:8080`.
```

File: website/docs/docs/core/connect-data-platform/snowflake-setup.md [hunk 5]
```diff
@@ -421,6 +421,30 @@ my-snowflake-db:
...
+
+Once your Snowflake admin has configured OAuth, add the following to your `profiles.yml` file:
```
Comment (runleonarun) [inline website/docs/docs/core/connect-data-platform/snowflake-setup.md:430]: I wonder if this and the previous should be steps?

**Review by runleonarun — APPROVED** [review]
small suggestion and quick question

File: website/docs/docs/core/connect-data-platform/snowflake-setup.md (hunk 5 above)
Comment (matthewshaver) [reply website/docs/docs/core/connect-data-platform/snowflake-setup.md:430]: This aligns with the examples in this portion of the docs. We rely on the adapter documentation for that. Important if it ever changes, then we don't have to do the heavy lifting.

File: website/docs/docs/core/connect-data-platform/snowflake-setup.md (hunk 4 above)
Comment (matthewshaver) [reply website/docs/docs/core/connect-data-platform/snowflake-setup.md]: They don't provide it in our product; it's in the Snowflake docs.

File: website/docs/docs/core/connect-data-platform/snowflake-setup.md (hunk 3 above)
Comment (matthewshaver) [inline website/docs/docs/core/connect-data-platform/snowflake-setup.md]:
```suggestion
To learn how to configure OAuth in Snowflake, refer to their [documentation](https://docs.snowflake.com/en/user-guide/oauth-snowflake-overview). Your Snowflake admin needs to generate an [OAuth token](https://community.snowflake.com/s/article/HOW-TO-OAUTH-TOKEN-GENERATION-USING-SNOWFLAKE-CUSTOM-OAUTH) for your configuration to work.
```

Comment (matthewshaver) [inline website/docs/docs/core/connect-data-platform/snowflake-setup.md]:
```suggestion
To learn how to configure OAuth in Snowflake, refer to their [documentation](https://docs.snowflake.com/en/user-guide/oauth-snowflake-overview). Your Snowflake admin needs to generate an [OAuth token](https://community.snowflake.com/s/article/HOW-TO-OAUTH-TOKEN-GENERATION-USING-SNOWFLAKE-CUSTOM-OAUTH) for your configuration to work.

```

**Pull request merged**